### PR TITLE
ARM64: Fix conversion double to unsigned int

### PR DESCRIPTION
**Pull request by kyulee1**
For ARM/ARM64, converting negative double to unsigned int is 0, which aligns with all native compilers.
I also removed the part of test that depends on architecture specific values.
Since ARM64 tests were prepopulated, I just updated the tag so that we can update the tests later.

### CODE REVIEW

**Comment by kyulee1** [comment]
@dotnet-bot test Windows_NT arm64

**Comment by kyulee1** [comment]
@dotnet/jit-contrib PTAL

File: src/jit/utils.cpp [hunk 1]
```diff
@@ -1604,6 +1605,10 @@ unsigned __int64 FloatingPointUtils::convertDoubleToUInt64(double d) {
     // is unsigned. gcc converts a negative double to zero when
     // cast to an unsigned. To make gcc conform to MSVC behavior
     // this cast is necessary.
-    u64 = UINT64(INT64(d));     
+    u64 = UINT64(INT64(d));
```
Comment (CarolEidt) [inline src/jit/utils.cpp]: Could you update this comment? I would say that the only "spec hole" is that it isn't clear that the "overflow" case would apply to converting a negative double to int.
I would reword this to say something like (also note that it should be "Ecma" not "ECMA"):

While the Ecma spec doesn't specifically call this out, the case of conversion from negative double to unsigned is effectively an overflow and therefore the result is unspecified.
With MSVC, such a conversion results in the bit-equivalent unsigned value of the conversion to integer.
Other compilers convert negative doubles to zero when the target is unsigned.
To make the behavior consistent across OS's on _TARGET_XARCH_, this double cast is needed to make gcc conform to MSVC behavior.

**Comment by CarolEidt** [comment]
LGTM with a suggested comment rewording.

File: src/jit/utils.cpp (hunk 1 above)
Comment (kyulee1) [inline src/jit/utils.cpp]: Thanks for the detailed comments. Will update it before merge.

**Comment by kyulee1** [comment]
Lab is down and the change won't affect any testing results that are covered. I'm just merging this.
